drivers/sensor/vl53l0x/vl53l0x.c

```diff
@@ -0,0 +1,254 @@
+/* vl53l0x.c - Driver for ST VL53L0X time of flight sensor */
+
+/*
+ * Copyright (c) 2017 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <kernel.h>
+#include <i2c.h>
+#include <sensor.h>
+#include <init.h>
+#include <gpio.h>
+#include <misc/__assert.h>
+
+#include "vl53l0x.h"
+
+static int vl53l0x_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct vl53l0x_data *drv_data = (struct vl53l0x_data *)dev->driver_data;
+	u8_t ret = 0;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL
+			|| chan == SENSOR_CHAN_DISTANCE
+			|| chan == SENSOR_CHAN_PROX);
+
+	ret = VL53L0X_PerformSingleRangingMeasurement(&drv_data->vl53l0x,
+					&drv_data->RangingMeasurementData);
+	if (ret < 0) {
+		SYS_LOG_ERR("Could not perform measurment (error=%d)", ret);
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+
+static int vl53l0x_channel_get(struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct vl53l0x_data *drv_data = (struct vl53l0x_data *)dev->driver_data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_DISTANCE
+			|| chan == SENSOR_CHAN_PROX);
+
+	val->val1 = drv_data->RangingMeasurementData.RangeMilliMeter;
+	val->val2 = 0;
```

Comment:
hum, I guess you need to put these 2 lines in an else after if (chan == SENSOR_CHAN_PROX) {}

It's useless to assign val1 if it's meant to be overwritten in case of chan being SENSOR_CHAN_PROX